public/app/features/alerting/unified/utils/datasource.ts

```diff
@@ -35,21 +35,33 @@ export function getLotexDataSourceByName(dataSourceName: string): DataSourceInst
   return dataSource;
 }
 
-export function isCloudRulesSource(rulesSource: RulesSource): rulesSource is DataSourceInstanceSettings {
+export function getAllRulesSourceNames(): string[] {
+  return [...getRulesDataSources().map((r) => r.name), GRAFANA_RULES_SOURCE_NAME];
+}
+
+export function getAllRulesSources(): RulesSource[] {
+  return [...getRulesDataSources(), GRAFANA_RULES_SOURCE_NAME];
+}
+
+export function getRulesSourceName(rulesSource: RulesSource): string {
+  return isCloudRulesSource(rulesSource) ? rulesSource.name : rulesSource;
+}
+
+export function isCloudRulesSource(rulesSource: RulesSource | string): rulesSource is DataSourceInstanceSettings {
   return rulesSource !== GRAFANA_RULES_SOURCE_NAME;
 }
 
 export function getDataSourceByName(name: string): DataSourceInstanceSettings<DataSourceJsonData> | undefined {
   return getAllDataSources().find((source) => source.name === name);
 }
 
-export function getDatasourceAPIId(dataSourceName: string): string {
-  if (dataSourceName === GRAFANA_RULES_SOURCE_NAME) {
+export function getDatasourceAPIId(datasourceName: string) {
```

Comment:
`dataSource` in this file 
 (sorry)

Reply:
oh no, a regression :laughing: 
- [ ] fix